### PR TITLE
Bug/tooltip

### LINKED ISSUE #248: Tooltip: unnecessary focus box when toggled
**Describe the bug**
Whenever the Tooltip component is toggled, there will be a focus highlight box surrounding the component.

**To Reproduce**
1. Run the docs
2. Visit Tooltip page
3. Scroll down to examples
4. Toggle the tooltip
5. See the issue

**Expected behavior**
The tooltip box should not have any surrounding highlight even if it's in focus. If the intended behavior is to toggle the tooltip on focus then the tooltip box shouldn't be the one to be highlighted. It will be the element that is the target of the tooltip. For example, an input field that comes in focus would toggle a tooltip.

**Screenshots**
Screen Shot 2020-04-09 at 2 11 51 PM

**Desktop (please complete the following information):**
 - OS: MacOS, Windows 10
 - Browser Chrome, Safari, Firefox, Edge, IE

### LINKED ISSUE #249: Tooltip: the container should not be translucent
**Describe the bug**
The tooltip container is currently translucent which it shouldn't be. There is no mention anywhere in the [Design Library Guidelines](https://designlibrary.sebgroup.com/patterns/feedback-to-the-user/guide-the-user/tooltip/#guidelines) of a tooltip container being translucent

**To Reproduce**
1. Run the docs
2. Visit Tooltip page
3. Scroll down to examples
4. Toggle the tooltip
6. Scroll the page where an item can come underneath it.
5. See the issue

**Expected behavior**
The tooltip container should be completely solid. No transparency is required.

**Screenshots**
Screen Shot 2020-04-09 at 2 29 14 PM

**Desktop (please complete the following information):**
 - OS: MacOS, Windows 10
 - Browser Chrome, Safari, Firefox, Edge, IE

### LINKED ISSUE #251: Tooltip: the container is still rendered even when it's hidden
**Describe the bug**
The tooltip container lingers around for some reason even though it's hidden

**To Reproduce**
1. Run the docs
2. Visit Tooltip page
3. Scroll down to examples
4. Switch to **Bottom** placement
5. Toggle on the tooltip with a click on the icon
6. Toggle off the tooltip with a click on the icon
7. Move a little bit underneath the tooltip and click
8. See the issue

**Expected behavior**
The tooltip should not be rendered when it's supposed to be hidden

**Screenshots**
Screen-Recording-2020-04-09-at-2 47 22-PM

**Desktop (please complete the following information):**
 - OS: MacOS, Windows 10
 - Browser Chrome, Safari, Firefox, Edge, IE

### PR DESCRIPTION
**Pull request by kherP**
fix #248 
fix #249 
fix #251

### CODE REVIEW

File: src/__utils/Overlay/Overlay.tsx [hunk 1]
```diff
@@ -46,7 +46,11 @@ const Overlay: React.FC<React.PropsWithChildren<OverlayProps>> = (props: React.P
     const onScroll = React.useCallback((ev: Event): void => {
         const target: HTMLDivElement = ev.target as HTMLDivElement;
         if (props.show && target.contains(props.overlayReference())) {
-            setPlacementWithCoords(tooltipPositionChecker.getPosition(props.position || "top"));
+            const referenceDomRect: DOMRect = props.overlayReference().getBoundingClientRect();
+            if (referenceDomRect.bottom < 0 || referenceDomRect.right < 0 || referenceDomRect.left > window.innerWidth || referenceDomRect.top > window.innerHeight) {
+                overlayContentRef.current.blur();
```
Comment (DanSallau) [inline src/__utils/Overlay/Overlay.tsx:51]:
```suggestion
                overlayContentRef?.current?.blur();
```

File: src/Tooltip/tooltip-style.scss [hunk 2]
```diff
@@ -46,6 +45,9 @@ $tooltip-distance: 4px;
                 vertical-align: top;
             }
         }
+        &.cursor {
```
Comment (yousifalraheem) [inline src/Tooltip/tooltip-style.scss:48]: This class name is too generic, can you use `cursor-pointer` instead? Just to be more specific

File: src/__utils/Overlay/overlay-style.scss [hunk 3]
```diff
@@ -1,6 +1,7 @@
 .overlay-container {
     position: absolute;
-    z-index: 1;
+    outline: none;
+    z-index: 9999;
```
Comment (yousifalraheem) [inline src/__utils/Overlay/overlay-style.scss:4]: That's too much `z-index`. Just use the variables defined in bootstrap:
```scss
$zindex-dropdown-backdrop:          990 !default;
$zindex-dropdown:                   1000 !default;
$zindex-sticky:                     1020 !default;
$zindex-fixed:                      1030 !default;
$zindex-modal-backdrop:             1040 !default;
$zindex-modal:                      1050 !default;
$zindex-popover:                    1060 !default;
$zindex-tooltip:                    1070 !default; // <-- This one
```

**Comment by yousifalraheem** [comment]
@kherP Please make sure you run `npm install` since a lot of the dependencies have changed.

**Pull request closed without merging**